New package: AstroBase v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: AstroBase
- Repository: https://github.com/GenAstro/Epicycle
- Created by: @stevenhughes73
- Version: v0.1.0
- Commit: 49b64883ffb8bf4be214ea94512aba08c5e83425
- Reviewed by: @stevenhughes73
- Reference: https://github.com/GenAstro/Epicycle/commit/49b64883ffb8bf4be214ea94512aba08c5e83425#commitcomment-170224033
- Description: An application for space systems with a nod to the giants who came before us and to how far we’ve come. 
- Release notes:
<!-- BEGIN RELEASE NOTES -->
`````
AstroBase is part of the Epicycle space mission design and navigation system.
Initial release - core abstract types and type hierarchy for astrodynamics.

This package is part of a larger monorepo. I am creating PRs for all packages with no deps or compat sections,
so import tests are expected to fail. After a three-day wait, I will retrigger registration from main.
This is a workaround to register interdependent packages in a monorepo in parallel.

pkg-registration thread is here: 
https://julialang.slack.com/archives/C6M4DQA5P/p1762795703772249?thread_ts=1762515100.591929&cid=C6M4DQA5P
`````
<!-- END RELEASE NOTES -->

<!-- 80cc8f6b60f354a8c40306cc5f8995f359b48e153b97e32d462378611f2709bca91997c38a2113b2a1790a85a0a87bf7a3d251dfb915b14a9734633a44fe19e1c1c93b2cf99a6ebf8e7645ad1b8b9a750470e82d1a6d218b9dcad18caa3045723359546fb47eb06379af76c29128790981c599c71dff425bbee19f706ae415d7a2dcb7c6ee9a7b3f3d214eb037d330525ce8332a4ccf3aa0af96642a4bd854f304b86c9e852da1cf0886e32f763c93c48220e398506e8f0a4439528852f949fa010e11246bd7e2b38d827fc5f0c6afe9af8e62ac511f79b85dd37bdabedeec60 -->

aplavin commented:
>     module AstroBase
> 
> Core abstract types shared across Epicycle (variables, states, controls, time, functions, points).

Seems like a super-generic name for such a package, maybe consider `EpicycleBase`?

JuliaRegistrator committed:
New package: AstroBase v0.1.0

UUID: cbc97457-4cae-4fc4-be5b-ffc98fbd0dcb
Repo: https://github.com/GenAstro/Epicycle.git
Tree: ae6c7ff83a42bc3872c84763045cd658d977a638

Registrator tree SHA: 50f504d641745716a5b3eabaf681d3a4937d2ae3
